solutions/platinum/cses-2429.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: cses-2429
+source: CSES
+title: Grid Completion
+author:
+---
+
+# Solution
+
+Time complexity: $\mathcal O(N^3)$
+
+<LanguageSection>
+
+<CPPSection>
+
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+typedef long long ll;
+const ll MOD = 1e9 + 7;
+const int maxN = 505;  // Maximum grid size
+
+int N, p[maxN], q[maxN], C[6];
+bool inp[maxN], inq[maxN];
+char S[maxN];
+ll fact[maxN], inv[maxN];
+
+// Compute modular inverse using Fermat's Little Theorem
+ll inverse(ll x) {
+	ll res = 1;
+	ll b = MOD - 2;  // Theorem: x^(MOD-2) ≡ x^(-1) (mod MOD)
+	while (b) {
+		if (b & 1) res = (res * x) % MOD;
+		x = (x * x) % MOD;
+		b >>= 1;
+	}
+	return res;
+}
+
+// Function to compute binomial coefficient C(x, y) modulo MOD
+ll choose(int x, int y) { return (fact[x] * inv[y] % MOD) * inv[x - y] % MOD; }
+
+// Precompute factorials and their modular inverses
+void init() {
+	fact[0] = inv[0] = 1;
+	for (int i = 1; i <= N; i++) {
+		fact[i] = (fact[i - 1] * i) % MOD;
+		inv[i] = (inv[i - 1] * inverse(i)) % MOD;
+	}
+}
+
+// Function to calculate the number of ways to complete the grid based on
+// parameters i, j, k
+ll f(int i, int j, int k) {
+	ll res = (choose(C[0], i) * choose(C[1], j) % MOD * choose(C[2], k) % MOD *
+	          choose(C[3], i) % MOD);
+
+	// Multiply by the factorials of the number of rows and columns left
+	res = (res * fact[i] % MOD * fact[C[4] - i - j] % MOD * fact[C[5] - i - k] %
+	       MOD);
+
+	// Apply inclusion-exclusion principle to adjust the result
+	if ((i + j + k) % 2 == 1) res = (MOD - res);
+
+	return res;
+}
+
+int main() {
+	scanf("%d", &N);
+
+	init();
+
+	// Read grid and fill arrays p, q, inp, and inq
+	for (int i = 0; i < N; i++) {
+		scanf(" %s", S);
+		p[i] = q[i] = -1;
+		for (int j = 0; j < N; j++) {
+			if (S[j] == 'A') {
+				p[i] = j;
+				inp[j] = true;
+			}
+			if (S[j] == 'B') {
+				q[i] = j;
+				inq[j] = true;
+			}
+		}
+	}
+
+	// Count various configurations based on positions of 'A' and 'B'
+	for (int i = 0; i < N; i++) {
+		if (p[i] == -1 && q[i] == -1) C[0]++;
```

Comment:
put braces around if statements

e.g. `if (p[i] == -1 && q[i] == -1) { C[0]++; }`

Reply:
(this applies to most things)